internal/agent/transport/grpc.go

```diff
@@ -0,0 +1,170 @@
+package transport
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/avast/retry-go"
+	"github.com/go-logr/logr"
+	"github.com/tinkerbell/tink/internal/agent/event"
+	"github.com/tinkerbell/tink/internal/agent/workflow"
+	workflowproto "github.com/tinkerbell/tink/internal/proto/workflow/v2"
+)
+
+var _ event.Recorder = &GRPC{}
```

Comment:
should we add a check that `&GRPC{}` implements the [`agent.Transport`](https://github.com/tinkerbell/tink/blob/29052da86f6551ce443a7c3a434da523543da389/internal/agent/transport.go#L10) interface too?

Reply:
Unfortunately that creates an import cycle - pretty common when locating interfaces next to the consumer.